save-orchestrator/src/main/kotlin/org/cqfn/save/orchestrator/service/DockerService.kt

```diff
@@ -80,17 +81,27 @@ class DockerService(private val configProperties: ConfigProperties) {
 
     /**
      * @param agentIds list of IDs of agents to stop
+     * @return true if agents have been stopped, false if another thread is already stopping them
      */
-    fun stopAgents(agentIds: List<String>) {
-        if (isAgentStoppingInProgress.compareAndSet(false, true)) {
-            agentIds.forEach {
-                log.info("Stopping agent with id=$it")
-                containerManager.dockerClient.stopContainerCmd(it).exec()
-                log.info("Agent with id=$it has been stopped")
+    fun stopAgents(agentIds: List<String>) =
+            if (isAgentStoppingInProgress.compareAndSet(false, true)) {
+                try {
+                    agentIds.forEach {
+                        log.info("Stopping agent with id=$it")
+                        containerManager.dockerClient.stopContainerCmd(it).exec()
+                        log.info("Agent with id=$it has been stopped")
+                    }
+                    true
+                } catch (dex: DockerException) {
+                    log.error("Error while stopping agents $agentIds", dex)
+                    false
```

Comment:
Do we need to do something else to stop container?

Reply:
There are two documented types of `DockerException` here: 
```java
     * @throws NotFoundException
     *             No such container
     * @throws NotModifiedException
     *             Container already stopped
```
Both of them seem to indicate that we shouldn't try to stop it again. And even if the container haven't been stopped, it will continue to send heartbeats and we will continue to try to stop it.